### PR TITLE
HZ-636 Do not create map config with phone home service

### PR DESCRIPTION
**Pull request by ahmetmircik**
Happens when phone home is enabled.

Closes: https://github.com/hazelcast/hazelcast/issues/19137
Backport of: https://github.com/hazelcast/hazelcast/pull/19903

### CODE REVIEW

**Comment by zhmaeff** [comment]
@erosb should this fix be also backported to 4.0.z 4.1.z 4.2.z as you did for https://github.com/hazelcast/hazelcast/issues/19833 ?

**Comment by erosb** [comment]
@zhmaeff @ahmetmircik yes please do raise backport PRs. Thanks a lot for the fix.

**Review by erosb — APPROVED** [review]

File: hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/MapInfoCollectorTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.junit.runner.RunWith;
+
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class})
```
Comment (vbekiaris) [inline hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/MapInfoCollectorTest.java:36]: nit: can the test be also `ParallelJVMTest` so it is executed in the parallel test execution phase?

**Review by vbekiaris — APPROVED** [review]
@ahmetmircik thanks for the fix.

File: hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/MapInfoCollectorTest.java (hunk 1 above)
Comment (zhmaeff) [reply hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/MapInfoCollectorTest.java:36]: Also nit: this test could be merged with `com.hazelcast.internal.util.phonehome.GhostMapTest`.

**Review by zhmaeff — APPROVED** [review]

**Review by vbekiaris — APPROVED** [review]

**Comment by zhmaeff** [comment]
@ahmetmircik thank you for this fix! Pls do not forget to raise backports for `4.0.z` and `4.1.z`.

**Comment by ahmetmircik** [comment]
@zhmaeff Can you verify the issue exists on that branches also?

**Comment by zhmaeff** [comment]
@ahmetmircik @erosb Oh, I thought it was already merged.

**Pull request merged**